src/main/java/io/cryostat/recordings/RecordingTargetHelper.java

```diff
@@ -156,6 +168,65 @@ public IRecordingDescriptor startRecording(
                 .findFirst();
     }
 
+    public Future<Optional<InputStream>> getRecording(
```

Comment:
Please add a comment here with some link to `TargetConnectionManager#markConnectionInUse`. The returned `InputStream`, if any, is only readable while the remote connection remains open. Callers need to be aware of this so that they don't try to read from the stream after the connection has been closed due to inactivity, or so that the TCM doesn't close the stream while it's still being consumed without the TCM realizing it.